torch/fx/traceback.py

```diff
@@ -15,12 +20,76 @@
     "format_stack",
     "set_current_meta",
     "get_current_meta",
+    "NodeSource",
+    "get_graph_provenance_json",
 ]
 
 current_meta: Dict[str, Any] = {}
 should_preserve_node_meta = False
 
 
+class NodeSource:
+    """
+    NodeSource is a data structure that contains the provenance information of a node.
+    If node `a` is created from node `b`, then `a.meta["from_node"]` may contain NodeSource(b).
+    """
+
+    node_name: str
```

Comment:
why not just `name`?

Reply:
> why not just `name`?

I could also just make it name. Just there's also a "pass_name". So I thought it might be good to specify that there are two types of names explicitly, node_name and pass_name.

Also `node_name` might not be available. In inductor (will be added in follow-up Diffs), sometimes we don't have a specific node source, but only pass_name. For example, in some graph pass (not an Interpreter pass, but a pre_grad or post_grad inductor graph pass), we made some changes. We might not know what is the specific source node, but at least we know which pass added this node. So `node_name` might be empty. 


Do you think just `name` is better?

Reply:
Yeah. Also if the node specific info is missing I wonder if it would be better to add another level of indirection instead of filling in defaults like `-1`, `""`, etc .